Silicon/CommonSocPkg/Library/IgdOpRegionLib/IgdOpRegionLib.c

```diff
@@ -206,10 +205,9 @@ IgdOpRegionInit (
   SetMem(mIgdOpRegion.OpRegion, sizeof(IGD_OPREGION_STRUCTURE) + ExtendedVbtSize, 0);
 
   //
-  // Update OpRegion address to Gnvs
+  // Update OpRegion address to Gnvs and Return back IMON current value
   //
-  Gnvs = (GLOBAL_NVS_AREA *)(UINTN)PcdGet32(PcdAcpiGnvsAddress);
-  Gnvs->SaNvs.IgdOpRegionAddress = (UINT32)(UINTN)(mIgdOpRegion.OpRegion);
+  GfxTurboIMON = UpdateIgdOpGnvs((UINT32)(UINTN)(mIgdOpRegion.OpRegion));

```

Comment:
This forces all platforms to have UpdateIgdOpGnvs(), so this is not standalone library. I think a parameter of IgdOpRegionInit() would be better - structure type parameter for future extension. Otherwise, it could be done in GopVbtUpdateCallback.

Reply:
Hi Aiden,
I did come checking, yes agreed with you, could put GVNS as an input params, as this is same for almost all platforms. In that case, i will move IgdOpRegionInit() into PlatformUpdateAcpiGnvs() instead, and i checked PlatformUpdateAcpiGnvs is called before IgdOpRegionInit, both are in Prepayload stage.

Reply:
Thanks for your review, good idea, add 1 extra function in every platform seems bit redundant also. I would prefer to pass in a struct table and point to that table from common codes, and a struct can be very flexible, and it is very useful when we make this common for CML, CFL & APL also (i did some comparison). WOuld it be possible to merge this in first, then i come out with improvements & include CML and CFL in v2 also? Since EHL is pending this patch.

Reply:
A single parameter would be enough.
```
typedef struct {
  UINTN data1;
  UINTN data2;
  ....
  GOP_VBT_UPDATE_CALLBACK callback;
} IGD_OP_PLATFORM_INFO;

IgdOpRegionInit (IGD_OP_PLATFORM_INFO *PlatformInfo)
```

Reply:
Thanks @myguitar! Done, hope it meet good coding standard :)